text/trace/0174-http-semantic-conventions.md

```diff
@@ -0,0 +1,166 @@
+and thus to ship and use stable instrumentation.
+
+> NOTE. This OTEP captures a scope for changes should be done to existing
+experimental semantic conventions for HTTP, but does not propose solutions.
+
+## Roadmap for v1.0
+
+1. This OTEP, consisting of scenarios/open questions and a proposed roadmap, is
+   approved and merged.
+2. [Stability guarantees](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#not-defined-semantic-conventions-stability)
+   for semantic conventions are approved and merged. This is not strictly related
+   to semantic conventions for HTTP but is a prerequisite for stabilizing any
+   semantic conventions.
+3. Separate PRs addressing the scenarios and open questions listed in this
+   document are approved and merged.
+4. Proposed specification changes are verified by prototypes for the scenarios
+   and examples below.
+5. The [specification for HTTP semantic conventions for tracing](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md)
+   are updated according to this OTEP and are declared
+   [stable](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#stable).
+
+The steps in the roadmap don't necessarily need to happen in the given order,
+some steps can be worked on in parallel.
+
+## Scope for v1.0: scenarios and open questions
+
+> NOTE. The scope defined here is subject for discussions and can be adjusted
+  until this OTEP is merged.
+
+Scenarios and open questions mentioned below must be addressed via separate PRs.
+
+### Required attribute sets
+
+> At least one of the following sets of attributes is required:
+>
+> * `http.url`
+> * `http.scheme`, `http.host`, `http.target`
+> * `http.scheme`, [`net.peer.name`](span-general.md), [`net.peer.port`](span-general.md), `http.target`
+> * `http.scheme`, [`net.peer.ip`](span-general.md), [`net.peer.port`](span-general.md), `http.target`
+
+As a result, users that write queries against raw data or Zipkin/Jaeger don't
+have consistent story across instrumentations and languages. e.g. they'd need to
+write queries like
+`select * where (getPath(http.url) == "/a/b" || getPath(http.target) == "/a/b")`
+
+### Retries and redirects
+
+Each try/redirect/hedging request must have unique context to be traceable and
+to unambiguously ask for support from downstream service, which implies span
+per call.
+
+Redirects: users may need observability into what server hop had an error/took
+too long. E.g., was 500/timeout from the final destination or a proxy?
+
+### Context propagation needs explanation
+
+* Reusing instances of client HTTP requests between tries (it’s likely, so clean
+  up context before making a call).
+
+### Security concerns
```

Comment:
Discussed in the instrumentation SIG: To what extent do we address this for v1?

Reply:
For v1 we can at least identify which attributes can potentially contain sensitive information and mark them accordingly. Added this to the OTEP.

Reply:
Potentially this should be out of scope for v1, on 2nd thoughts. Addressing security should be thought through very carefully, so maybe it should be addressed separately.

Reply:
100%. v1 doesn't need to be feature rich, just useable and reliable. We can fast follow with incremental additions.

Reply:
Discussed this at Instrumentation SIG meeting on Nov 30 (https://docs.google.com/document/d/1dWHhyXnfVife-cQ2DW5-d5Ldp1Lq8Rre2UsHpyo8cEE/edit#):
![image](https://user-images.githubusercontent.com/5635171/144318091-0db006eb-5c47-4a14-87ec-b155e3490216.png)

It seems important for v1 to at least identify attributes that can potentially contain sensitive information, so customers can setup in-proc or out-of-proc processing of such attributes based on their knowledge of the domain.

Going forward, we can incrementally add more details on how exactly that data could be processed/sanitized.